src/clusterserialize.jl

```diff
@@ -167,10 +167,12 @@ function deserialize_global_from_main(s::ClusterSerializer, sym)
             return nothing
         end
     end
-    Core.eval(Main, Expr(:global, sym))
     if sym_isconst
-        ccall(:jl_set_const, Cvoid, (Any, Any, Any), Main, sym, v)
+        Core.eval(Main, Expr(:block,
+            Expr(:const, GlobalRef(Main, sym), v),
+            nothing))
```

Comment:
What is the `nothing` argument to Expr for?

Reply:
The post-lowering const form is not allowed in value position, so there needs to be a dummy nothing to drop the return value

Reply:
Would it be helpful to add a code comment with the above info?

Reply:
Yeah if we could add that as a comment before merging that'd be great.

Reply:
(took the liberty of adding it in 52d5703)